flood_adapt/dbs_classes/dbs_template.py

```diff
@@ -0,0 +1,370 @@
+        """
+
+        # Check if object is a standard object
+        self._check_standard_objects(name)
+
+        # Check if object is used in a higher level object
+        used_in = self._check_higher_level_usage(name)
+
+        # Check if the object is locked by another process
+        if self.is_locked(name=name):
+            raise ValueError(
+                f"'{name}' {self._type} is locked by another process and cannot be deleted."
+            )
+
+        # If measure is used in a strategy, raise error
+        if used_in:
+            raise ValueError(
+                f"'{name}' measure cannot be deleted since it is already used in: {', '.join(used_in)}"
+            )
+
+        # Once all checks are passed, delete the object
+        path = self._path / name
+        if toml_only:
+            # Only delete the toml file
+            toml_path = path / f"{name}.toml"
+            if toml_path.exists():
+                toml_path.unlink()
+            # If the folder is empty, delete the folder
+            if not list(path.iterdir()):
+                path.rmdir()
+        else:
+            # Delete the entire folder
+            shutil.rmtree(path, ignore_errors=True)
+
+    def _check_standard_objects(self, name: str):
+        """Checks if an object is a standard object.
+
+        Parameters
+        ----------
+        name : str
+            name of the object to be checked
+        """
+        # If this function is not implemented for the object type, it cannot be a standard object.
+        # By default, do nothing
+        pass
+
+    def _check_higher_level_usage(self, name: str) -> list[str]:
+        """Checks if an object is used in a higher level object.
+
+        Parameters
+        ----------
+        name : str
+            name of the object to be checked
+
+        Returns
+        -------
+        list[str]
+            list of higher level objects that use the object
+        """
+        # If this function is not implemented for the object type, it cannot be used in a higher
+        # level object. By default, return an empty list
+        return []
+
+    def _get_object_list(self) -> dict[Path, datetime]:
+        """Given an object type (e.g., measures) get a dictionary with all the toml paths
+        and last modification dates that exist in the database.
+
+        Returns
+        -------
+        dict[str, Any]
+            Includes 'path' and 'last_modification_date' info
+        """
+        base_path = self.input_path / self._folder_name
+        directories = list(base_path.iterdir())
+        paths = [Path(dir / f"{dir.name}.toml") for dir in directories]
+        last_modification_date = [
+            datetime.fromtimestamp(file.stat().st_mtime) for file in paths
+        ]
+
+        objects = {
+            "path": paths,
+            "last_modification_date": last_modification_date,
+        }
+
+        return objects
```

Comment:
might be nice to return the paths + the last_modfication_date as a key value pair in the objects dict.
something like: `objects = {path: date for path, date in zip(paths, last_modification_date)}`

Reply:
The idea of the function is that you can easilly get a list of all paths and in this case also the dates (not sure if we use them anywhere). If you look at where this function is called, this dict will then be extended with a list of names, descriptions e.d.